src/tests.rs

```diff
@@ -191,17 +194,60 @@ mod buf {
             buf.put_prefix_varint(*n);
         }
         let mut result = vec![];
-        let mut decode_data = buf.as_slice();
+        let decode_data = buf.as_slice();
         for decoded in decode_data.iter_prefix_varint::<i16>() {
             result.push(decoded.unwrap());
         }
         assert_eq!(to_encode, result.as_slice());
     }
 
+    #[test]
+    fn iterator_on_vec_works() {
+        let to_encode = [1, 2, -30, -24_000];
+        let mut buf = vec![];
+        for n in to_encode.iter() {
+            buf.put_prefix_varint(*n);
+        }
+        let mut result = vec![];
+        for decoded in buf.iter_prefix_varint::<i16>() {
+            result.push(decoded.unwrap());
+        }
+    }
+
+    #[test]
+    fn iterator_on_vec_deq_works() {
+        let to_encode = [1, 2, -30, -24_000];
+        let mut buf = vec![];
+        for n in to_encode.iter() {
+            buf.put_prefix_varint(*n);
+        }
+        let mut result = vec![];
+        let buf = VecDeque::from(buf);
+        for decoded in buf.iter_prefix_varint::<i16>() {
+            result.push(decoded.unwrap());
+        }
+    }
+
+    #[test]
+    fn iterator_on_bytes_buf_works() {
+        let to_encode = [1, 2, -30, -24_000];
+        let mut buf = vec![];
+        for n in to_encode.iter() {
+            buf.put_prefix_varint(*n);
+        }
+        let mut result = vec![];
+        bytes::Bytes::from(buf)
```

Comment:
do the same `.collect::<Result<Vec<_>, _>()` trick I mentioned above here.

Reply:
Roger, changed. One thing to note is that `VecDeq` does not have an efficient implementation for `Buf` being consumed because it can not deref into a `&[u8]`. So this implementation consumes `VecDeq` while it does not consume `Vec<u8>`